### PR TITLE
feat: script runner part 1

### PR DESCRIPTION
**Pull request by YazeedLoonat**
## Issue Overview

This PR addresses [#issue](https://github.com/bloom-housing/bloom/issues/4079)

- [x] This change addresses only certain aspects of the issue

## Description
This creates the framework for running scripts including:
- the controller
- an example end point
- helper functions for managing the state of the script
- the service
- an example script to run

## How Can This Be Tested/Reviewed?
Unit tests were added, but I don't think there is really much to tests here locally as this is just the first part with set up

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [x] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: api/src/services/script-runner.service.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+/**
+  this is the service for running scripts
+  most functions in here will be unique, but each function should only be allowed to fire once
+*/
+@Injectable()
+export class ScriptRunnerService {
+  constructor(private prisma: PrismaService) {}
+
+  /**
+    this is simply and example
```
Comment (mcgarrye) [inline api/src/services/script-runner.service.ts:17]: nit: `simply an example`

File: api/src/services/script-runner.service.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    return { success: !!rawJurisdictions.length };
+  }
+
+  // |------------------ HELPERS GO BELOW ------------------ | //
+
+  /**
+   *
+   * @param scriptName the name of the script that is going to be run
+   * @param userTriggeringTheRun the user that is attempting to trigger the script run
+   * @description this checks to see if the script has already ran, if not marks the script in the db
```
Comment (mcgarrye) [inline api/src/services/script-runner.service.ts:33]: What if we intentionally want a script to be run multiple times?

File: api/prisma/migrations/20240517175607_add_script_runs/migration.sql [hunk 3]
```diff
@@ -0,0 +1,14 @@
+-- CreateTable
```
Comment (mcgarrye) [inline api/prisma/migrations/20240517175607_add_script_runs/migration.sql:1]: File name is a date not a number

File: api/src/services/script-runner.service.ts (hunk 2 above)
Comment (YazeedLoonat) [reply api/src/services/script-runner.service.ts:33]: I think we should put some friction around running a script multiple times

we always have the option of deleting the script run entry in the db and running again. But I think most scripts should be considered fire once

if we really needed to we could do some stuff with creating multiple endpoints and each endpoint sends a unique script name to the same function in the service so I do think that option is still available

File: api/prisma/migrations/20240517175607_add_script_runs/migration.sql (hunk 3 above)
Comment (YazeedLoonat) [reply api/prisma/migrations/20240517175607_add_script_runs/migration.sql]: fixed

File: api/src/services/script-runner.service.ts (hunk 1 above)
Comment (YazeedLoonat) [reply api/src/services/script-runner.service.ts]: updated

**Review by mcgarrye — APPROVED** [review]

**Review by ColinBuyck — APPROVED** [review]
LGTM 🌔 🥣

**Pull request merged**